x-pack/plugins/security_solution/server/lib/timeline/routes/README.md

```diff
@@ -323,4 +323,292 @@ kbn-version: 8.0.0
+
+kbn-version: 8.0.0
+
+```
+
+##### Request param
+
+timelineType `default` or `template`
+
+
+##### Response
+```json
+{
+    "data": {
+        "persistTimeline": {
+            "timeline": {
+                "savedObjectId": "ababbd90-99de-11ea-8446-1d7fd9f03ebf",
+                "version": "WzM2MiwzXQ==",
+                "columns": [
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "@timestamp"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "message"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "event.category"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "event.action"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "host.name"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "source.ip"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "destination.ip"
+                    },
+                    {
+                        "columnHeaderType": "not-filtered",
+                        "id": "user.name"
+                    }
+                ],
+                "dataProviders": [],
+                "description": "",
+                "eventType": "all",
+                "filters": [],
+                "kqlMode": "filter",
+                "timelineType": "default",
+                "kqlQuery": {
+                    "filterQuery": null
+                },
+                "title": "",
+                "sort": {
+                    "columnId": "@timestamp",
+                    "sortDirection": "desc"
+                },
+                "status": "draft",
+                "created": 1589899222908,
+                "createdBy": "casetester",
+                "updated": 1589899222908,
+                "updatedBy": "casetester",
+                "templateTimelineId": null,
+                "templateTimelineVersion": null,
+                "favorite": [],
+                "eventIdToNoteIds": [],
+                "noteIds": [],
+                "notes": [],
+                "pinnedEventIds": [],
+                "pinnedEventsSaveObject": []
+            }
+        }
+    }
+}
+```
+
+## Create draft timeline api
+
+#### POST /api/timeline/_draft
+
+##### Authorization
+
+Type: Basic Auth
+
+username: Your Kibana username
+
+password: Your Kibana password
+
+
+##### Request header
+
+```
+
+Content-Type: application/json
+
+kbn-version: 8.0.0
+
+```
+
+##### Request body
+
+```json
+{
+	"timelineType": "default" or "template"
```

Comment:
if i dont pass a body the error I get is:
`"[request body]: Invalid value \"null\" supplied to \"\""`
I wish it told me the key `timelineType`

Reply:
Thanks for pointing that out!
I have a PR up for fixing that: https://github.com/elastic/kibana/pull/65448
I'll make sure this won't happen after the fix.